_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,419 @@
+
+{% img blog/jhipster-digital-ocean/do-control-panel-token.png alt:"DigitalOcean Control Panel for Token Create" width:"500" %}{: .center-image }
+
+You can find a detailed list of pricing for cluster resources at [DigitalOcean](https://docs.digitalocean.com/products/kubernetes/),  and with `doctl` you can quickly retrieve a list of node size options available for your account:
+
+```bash
+doctl k options sizes
+```
+
+**NOTE**: I tested the cluster with the higher size Intel nodes available for my account, `s-2vcpu-4gb-intel`, in an attempt to run the application in the default cluster configuration, a three-node cluster with a single node pool in the nyc1 region, using the latest Kubernetes version. As I started to see pods not running due to "insufficient CPU", I increased the number of nodes later. DigitalOcean's latest and default Kubernetes version at the moment of writing this post is 1.22.8.
+
+Create the cluster with the following command line:
+
+```bash
+doctl k cluster create do1 -v --size s-2vcpu-4gb-intel
+```
+After creating a cluster, `doctl` adds a configuration context to kubectl and makes it active, so you can start monitoring your cluster right away with k9s. First, apply the resources configuration to the DigitalOcean cluster:
+
+```bash
+./kubectl-apply.sh -f
+```
+
+Monitor the deployment with k9s:
+
+```bash
+k9s -n demo
+```
+
+{% img blog/jhipster-digital-ocean/k9s-do-cluster.png alt:"k9s user interface monitoring DigitalOcean Kubernetes" width:"800" %}{: .center-image }
+
+Once you see the `jhipster-registry` pods are up, set up port forwarding again so you can also monitor the status of the services in the registry UI.
+
+### Increase CPU
+
+The three-node cluster might not provide enough CPU for this microservices architecture, so you might see `insufficient CPU` logs for some pods. You can get the pod events with `kubectl describe`:
+
+```bash
+kubectl describe pod jhipster-registry-0 -n demo
+```
+```text
+Events:
+  Type     Reason                  Age                   From                     Message
+  ----     ------                  ----                  ----                     -------
+  Normal   NotTriggerScaleUp       4m30s                 cluster-autoscaler       pod didn't trigger scale-up:
+  Warning  FailedScheduling        69s (x3 over 95s)     default-scheduler        0/3 nodes are available: 3 Insufficient cpu.
+```
+Increase the number of nodes with `doctl`:
+
+```bash
+doctl k cluster node-pool update do1 do1-default-pool --count 4
+```
+
+### Increase Storage
+
+For the `store-mongodb` stateful set with 3 replicas, some of the pods did not run due to _unbound immediate PersistentVolumeClaims_. When inspecting the pod events, I found out there was a failure in volume provisioning:
```

Comment:
How did you see these errors? I want to see if I'm experiencing the same thing?

Reply:
I realize I did not describe it well. Looking it my notes, with kubectl, I listed persistent volume claims `kubectl get pvc -n demo`, and then I did `kubectl describe pvc datadir-store-mongodb-2 -n demo` and saw the error in the events. I will clarify in the post.

Reply:
Added the commands to the post.